### PR TITLE
Contact filter for FRs on multidevice

### PR DESCRIPTION
**Pull request by vincentbavitz**
# Issues Fixed

## Message Failure Retry
A1 (master) and A2 (slave) are linked
A messages B1 (master) and B2 (slave)
If the send to B1 is successful, but the send to B2 isn't, we should throw an error.

## Contact Duplication
Fixes friend request to multi-device listing pubkeys of all devices under contacts.
When ``A1`` is linked to ``A2``, and B sends a FR to ``A1``, two FRs are listed in Contacts left pane.

Screenshot_20200504_132953

Screenshot_20200504_155024

## Friend Request Duplication
When A2 sends a friend request to B, and A1 sends a FR to B, before B accepts, B has two FRs displayed.

Screenshot_20200512_075238

### CODE REVIEW

File: ts/components/session/LeftPaneContactSection.tsx [hunk 1]
```diff
@@ -321,9 +321,15 @@ export class LeftPaneContactSection extends React.Component<Props, State> {
 
   private renderList() {
     const { sentFriendsRequest } = this.props;
+
+    const visibleFriendRequests = sentFriendsRequest.filter(
```
Comment (Bilb) [inline ts/components/session/LeftPaneContactSection.tsx:325]: Would be better to update the redux fetch directly (`sentFriendsRequest()` is not used anywhere else than in this file)
https://github.com/loki-project/session-desktop/blob/clearnet/ts/state/selectors/conversations.ts#L140

Also, if you change it here only, I think you have to update the `renderRow()` method too, no?

Comment (vincentbavitz) [reply ts/components/session/LeftPaneContactSection.tsx]: Fixed

File: ts/components/session/LeftPaneContactSection.tsx [hunk 2]
```diff
@@ -321,9 +321,12 @@ export class LeftPaneContactSection extends React.Component<Props, State> {
 
   private renderList() {
     const { sentFriendsRequest } = this.props;
+
     const friends = window.getFriendsFromContacts(this.props.friends);
     const length = Number(sentFriendsRequest.length) + Number(friends.length);
-    const combined = [...sentFriendsRequest, ...friends];
+
+    // Prevent where friends and send FR showing two entries
+    const combined = [...new Set([...sentFriendsRequest, ...friends])];
```
Comment (Bilb) [inline ts/components/session/LeftPaneContactSection.tsx:329]: not sure about how that can happen

Comment (vincentbavitz) [reply ts/components/session/LeftPaneContactSection.tsx:329]: Please see the .md

image-20200504133017652
image-20200512075350722

**Pull request merged**